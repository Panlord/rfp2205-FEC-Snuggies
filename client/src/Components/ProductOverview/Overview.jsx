import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { ProductOverviewContainer, ProductInformationDescription } from './StyledComponents/Containers.jsx';
import Gallery from './ImageGallery/Gallery.jsx';
import Information from './ProductInformation/Information.jsx';

var Overview = (props) => {

  const [product, setProduct] = useState({});
  const [styles, setStyles] = useState([]);
  const [chosenStyle, setChosenStyle] = useState({});
  const [reviews, setReviews] = useState({});

  // Upon component mounting, send a GET request and get all the relevant data (this may need to be refactored if the team decides to house all client-side request handling in the main App file)
  useEffect(() => {
    // Send axios request to the specific product
    axios.get('/snuggie/products', { params: { product_id: 40344 } }) // Right now, we have a placeholder for the specific product
      // Then set the products state
      .then((results) => {
        return setProduct(results.data);
      })
      // Send axios request to get all styles
      .then(() => {
        return axios.get('/snuggie/styles', { params: { product_id: 40344 } });
      })
      // Then set the styles state
      .then((results) => {
        return setStyles(results.data.results);
      })
      // Send axios request to get all the review data that is relevant for the specific product
      // Then set the reviews state
      .catch((error) => {
        console.log('An error occurred when initializing data received from server:', error);
      });
  }, []); // The second argument is an empty array, which makes it so that we only run the function once

  return (
    <div>
      <ProductOverviewContainer>
        <Gallery chosenStyle={chosenStyle} />
        <Information product={product} styles={styles} chosenStyle={chosenStyle}  setChosenStyle={setChosenStyle} reveiws={reviews} />
      </ProductOverviewContainer>
      <ProductInformationDescription>
        {product.description}
      </ProductInformationDescription>
    </div>
  );

}

export default Overview;