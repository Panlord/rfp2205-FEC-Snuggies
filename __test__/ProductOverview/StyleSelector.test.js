import React from 'react';
import renderer from 'react-test-renderer';
import Overview from '../../client/src/Components/ProductOverview/Overview.jsx';

it('changes the Main Image and thumbnails displayed when a different style is clicked', () => {
  // Render the Product Overview widget
  const Widget = renderer
    .create(<Overview />);
  new Promise(process.nextTick);
  let tree = Widget.toJSON();
  expect(tree).toMatchSnapshot();

  // Click on the second style thumbnail in the style selector
  renderer.act(() => {
    Widget.getInstance().findAllByType(StyleThumbnail)[1].props.onClick();
  });
  // Re-rendering
  tree = Widget.toJSON();
  expect(tree).toMatchSnapshot();

});