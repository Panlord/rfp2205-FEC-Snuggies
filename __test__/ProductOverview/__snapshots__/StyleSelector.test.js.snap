// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the Main Image and thumbnails displayed when a different style is clicked 1`] = `
<div>
  <section
    className="Containers__ProductOverviewContainer-sc-1le4a1r-0 ctVEHL"
  >
    <section
      className="Containers__ImageGalleryContainer-sc-1le4a1r-1 jDuarS"
    >
      <section
        className="ImageThumbnails-sc-lalhr2-0 jhuyUZ"
      />
      <div>
        <div
          className="MainImage__DefaultView-sc-k688fp-1 hoEhgE"
        >
          <div
            className="MainImage__Overlay-sc-k688fp-2 zLpAw"
          >
            <i
              className="MainImage__FullScreen-sc-k688fp-5 kGlmLV fa-solid fa-expand"
            />
          </div>
          <img
            className="MainImage__Image-sc-k688fp-0 ixsHNo"
            src=""
          />
        </div>
      </div>
    </section>
    <section
      className="Containers__ProductInformationContainer-sc-1le4a1r-2 hdNNLJ"
    >
      <section
        className="Containers__ProductOverviewStarContainer-sc-1le4a1r-3 iPUFAv"
      >
        <div>
          ⭐️⭐️⭐️⭐️⭐️
        </div>
        <span
          className="OverviewStars__ReviewsLink-sc-15f5p6d-0 cDYSFf"
        >
          Read [#] Reviews
        </span>
      </section>
      <div />
      <div />
      <div />
      <div>
        Style &gt; 
        Selected Style
      </div>
      <div>
        <section
          className="Containers__StyleThumbnailContainer-sc-1le4a1r-4 eahcuA"
        />
      </div>
      <div>
        <section
          className="Containers__SizeAndCountContainer-sc-1le4a1r-6 jhIKwT"
        >
          <select
            className="SizeDropdown-sc-vwdwce-0 iUhIvU"
          >
            <option
              value="S"
            >
              S
            </option>
            <option
              value="M"
            >
              M
            </option>
            <option
              value="L"
            >
              L
            </option>
          </select>
          <select
            className="QuantityDropdown-sc-2hwqqc-0 gCyTWg"
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
          </select>
        </section>
        <section
          className="Containers__AddToCartContainer-sc-1le4a1r-7 kzyAcV"
        >
          <button
            className="AddToCart__AddToCartButton-sc-1w0dmj9-0 ldXHUb"
          >
            Add to Cart
          </button>
          <button
            className="Favorite__FavoriteButton-sc-1czrrgn-0 eFRyuV"
          >
            ❤️ Button
          </button>
        </section>
      </div>
    </section>
  </section>
  <section
    className="Containers__ProductInformationDescription-sc-1le4a1r-8 dMtAPk"
  />
</div>
`;
